base/strings/search.jl

```diff
@@ -3,18 +3,39 @@
 const Chars = Union{Char,Tuple{Vararg{Char}},AbstractVector{Char},Set{Char}}
 
 """
-    search(string::AbstractString, chars::Chars, [start::Integer])
+    search(string, chars, [start::Integer])
 
 Search for the first occurrence of the given characters within the given string. The second
 argument may be a single character, a vector or a set of characters, a string, or a regular
 expression (though regular expressions are only allowed on contiguous strings, such as ASCII
-or UTF-8 strings). The third argument optionally specifies a starting index. The return
-value is a range of indexes where the matching sequence is found, such that `s[search(s,x)] == x`:
+or UTF-8 strings).
+Alternatively the first and the second arguments may be arrays of `Int8` or `UInt8`.
+The third argument optionally specifies a starting index.
+If it is not a valid index into the first argument then error is thrown.
+The return value when the second argument is a string or a vector of bytes is the samllest
+range of indexes where the matching sequence is found, such that `s[search(s,x)] == x`,
+or `0:-1` if no such range of indexes exits:
 
-`search(string, "substring")` = `start:end` such that `string[start:end] == "substring"`, or
-`0:-1` if unmatched.
+`search(string, "substring", i)` = `start:end` such that `string[start:end] == "substring"`
+and `start >= i`.
 
-`search(string, 'c')` = `index` such that `string[index] == 'c'`, or `0` if unmatched.
+In particular a search for a zero length second argument returns `i:(i-1)`.
+
+If the first argument is empty and third argument is not given `0:-1` is returned.
```

Comment:
Better give the general rule and use `start=start(s)` in the signature. People will figure out what happens in special cases.